Packages/io.chainsafe.web3-unity.web3auth/Runtime/Web3AuthConnectionProvider.cs

```diff
@@ -94,6 +99,17 @@ public override async Task Initialize(bool rememberSession)
         await _initializeTcs.Task;
     }
 #endif
+    
+#if UNITY_EDITOR
+    private void OnValidate()
```

Comment:
you could do this OnReset and avoid #if directive, plus happens when you create the SO only (once)

Reply:
Can move to OnReset you're right. Will have to keep the `#if` as it uses `UnityEditor` namespace.